add draft json schemas

halamix2 opened the pull request:
- #104

jonschz commented:
Just a suggestion: Have you looked into generating the JSON schemas from pydantic (https://docs.pydantic.dev/latest/concepts/json_schema/#JSON%20Schema)? If that works well enough, it would be preferable since then you'd have a single source of truth and wouldn't have to maintain the schema in two places.

halamix2 commented:
I haven't, I'll look into that

halamix2 commented:
I've added required changes to the `reccmp/project/config.py` file, and for now I've added `generate_schemas.py` and the generated files.
Now the questions are
- where to put the generated files
- and where to put the generator, should it run in `pip install`, should it run manually, or even as a push Github Action?
  - if run manualy should it be connected to a postcommit githook?
